iroh/src/magicsock/transports.rs

```diff
@@ -0,0 +1,432 @@
+        // but we never get data from both sockets at the same time in `poll_recv`
+        // and it's impossible and unnecessary to be refactored that way.
+
+        let res = self.ip.iter().map(|t| t.max_receive_segments()).max();
+        res.unwrap_or(1)
+    }
+
+    #[cfg(wasm_browser)]
+    pub(crate) fn max_receive_segments(&self) -> usize {
+        1
+    }
+
+    #[cfg(not(wasm_browser))]
+    pub(crate) fn may_fragment(&self) -> bool {
+        self.ip.iter().any(|t| t.may_fragment())
+    }
+
+    #[cfg(wasm_browser)]
+    pub(crate) fn may_fragment(&self) -> bool {
+        false
+    }
+
+    /// Check if a transport is writable, aka sendable on, for the given `addr`.
+    pub(crate) fn poll_writable(&self, cx: &mut Context, addr: &Addr) -> Poll<io::Result<()>> {
+        // TODO: what about multiple matches?
+        match addr {
+            #[cfg(wasm_browser)]
+            Addr::Ipv4(..) | Addr::Ipv6(..) => Poll::Ready(Err(io::Error::other(
+                "IP based addressing is not supported in the browser",
+            ))),
+            #[cfg(not(wasm_browser))]
+            Addr::Ipv4(..) | Addr::Ipv6(..) => {
+                let addr: SocketAddr = addr.clone().try_into().expect("known good");
+                match self.ip.iter().find(|t| t.is_valid_send_addr(&addr)) {
+                    Some(t) => t.poll_writable(cx),
+                    None => Poll::Pending,
+                }
+            }
+            Addr::RelayUrl(url, node_id) => {
+                match self
+                    .relay
+                    .iter()
+                    .find(|t| t.is_valid_send_addr(url, node_id))
+                {
+                    Some(t) => t.poll_writable(cx),
+                    None => Poll::Pending,
+                }
+            }
+        }
+    }
+
+    pub(crate) fn create_io_poller(&self) -> Pin<Box<dyn quinn::UdpPoller>> {
+        // To do this properly the MagicSock would need a registry of pollers.  For each
+        // node we would look up the poller or create one.  Then on each try_send we can
+        // look up the correct poller and configure it to poll the paths it needs.
+        //
+        // Note however that the current quinn impl calls UdpPoller::poll_writable()
+        // **before** it calls try_send(), as opposed to how it is documented.  That is a
+        // problem as we would not yet know the path that needs to be polled.  To avoid such
+        // ambiguity the API could be changed to a .poll_send(&self, cx: &mut Context,
+        // io_poller: Pin<&mut dyn UdpPoller>, transmit: &Transmit) -> Poll<io::Result<()>>
+        // instead of the existing .try_send() because then we would have control over this.
+        //
+        // Right now however we have one single poller behaving the same for each
+        // connection.  It checks all paths and returns Poll::Ready as soon as any path is
+        // ready.
```

Comment:
I'm not sure any longer how correct this comment is.  Maybe we should stop cargo-culting it around.  The issue as we currently understand it is really that quinn needs to be aware of multiple sockets so that it can understand pending differently for each one.  And less to do with per-NodeId stuff.

Reply:
so I should remove the comment?

Reply:
Yeah, I think so.

Reply:
done